File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.swift
import Foundation

func solution(_ my_string:String, _ queries:[[Int]]) -> String {
    
    var str = my_string
    
    for query in queries {
        let first = str.index(str.startIndex, offsetBy: query[0])
        let last = str.index(str.startIndex, offsetBy: query[1])
        str.replaceSubrange(first...last, with: str[first...last].reversed())
    }
    
    return str
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

string solution(string my_string, vector<vector<int>> queries) {
    for (auto query : queries) {
        int startIndex = query[0];
        int endIndex = query[1];
        string replaceStr = "";

        for (int i = endIndex; i >= startIndex; i--) {
            replaceStr += my_string[i];
        }

        my_string.replace(startIndex, endIndex - startIndex + 1, replaceStr);
    }

    return my_string;
}